control-plane-operator/hostedclusterconfigoperator/controllers/resources/kas/admissionpolicies.go

```diff
@@ -0,0 +1,128 @@
+package kas
+
+import (
+	"context"
+	"fmt"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/control-plane-operator/hostedclusterconfigoperator/controllers/resources/manifests"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/upsert"
+	k8sadmissionv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type AdmissionPolicyBlacklist struct {
+	AdmissionPolicies map[string]AdmissionPolicy
+}
+
+type AdmissionPolicy struct {
+	Name        string
+	APIGroup    []string
+	ApiVersion  []string
+	Resources   []string
+	Validations []k8sadmissionv1beta1.Validation
+}
+
+const (
+	AdmissionPolicyNameConfig = "config"
+	AdmissionPolicyNameICSP   = "icsp"
+)
+
+var (
+	defaultAdmissionPoliciesOperations = []k8sadmissionv1beta1.OperationType{"UPDATE", "DELETE"}
+	HCCOUserValidation                 = k8sadmissionv1beta1.Validation{
+		Expression: fmt.Sprintf("request.userInfo.username.contains('system:%s') || (has(object.spec) && object.spec == oldObject.spec)", config.HCCOUser),
```

Comment:
I don't think `has(object.spec)` is needed here. `spec` can never be nil.

Reply:
In the other tries, that. was failing because spec was nil, I think that's happened during 'CREATE'. Let me remove it.